packages/ui/src/utilities/buildTableState.ts

```diff
@@ -165,65 +164,14 @@ export const buildTableState = async (
   // get prefs, then set update them using the columns that we just received
   const preferencesKey = `${collectionSlug}-list`
 
-  const preferencesResult = await payload
-    .find({
-      collection: 'payload-preferences',
-      depth: 0,
-      limit: 1,
-      pagination: false,
-      where: {
-        and: [
-          {
-            key: {
-              equals: preferencesKey,
-            },
-          },
-          {
-            'user.relationTo': {
-              equals: user.collection,
-            },
-          },
-          {
-            'user.value': {
-              equals: user.id,
-            },
-          },
-        ],
-      },
-    })
-    .then((res) => res.docs[0] ?? { id: null, value: {} })
-
-  let newPrefs = preferencesResult.value
-
-  if (!preferencesResult.id || !dequal(columns, preferencesResult?.columns)) {
-    const preferencesArgs = {
-      collection: 'payload-preferences',
-      data: {
-        key: preferencesKey,
-        user: {
-          collection: user.collection,
-          value: user.id,
-        },
-        value: {
-          ...(preferencesResult?.value || {}),
-          columns,
-        },
-      },
-      depth: 0,
-      req,
-    }
-
-    if (preferencesResult.id) {
-      newPrefs = await payload
-        .update({
-          ...preferencesArgs,
-          id: preferencesResult.id,
-        })
-        ?.then((res) => res.value as ListPreferences)
-    } else {
-      newPrefs = await payload.create(preferencesArgs)?.then((res) => res.value as ListPreferences)
-    }
-  }
+  const preferencesResult = await payload.updatePreference({
```

Comment:
If the user says "give me columns A, B, and C" through the `buildTableState`, and their preferences are _already_ "A, B, and C" then we should not be updating here.

We may not be able to get away with simply calling `updatePreference` here - unless we made the internal `updatePreference` call smart enough to check to make sure they are different before actually running the `upsert`.